Source/JavaScriptCore/bytecode/CallLinkInfo.cpp

```diff
@@ -365,16 +356,11 @@ JSGlobalObject* CallLinkInfo::globalObjectForSlowPath(JSCell* owner)
 
 #if ENABLE(JIT)
 
-void OptimizingCallLinkInfo::setSlowPathCallDestination(CodePtr<JSEntryPtrTag> codePtr)
-{
-    m_slowPathCallDestination = codePtr;
-}
-
-std::tuple<MacroAssembler::JumpList, MacroAssembler::Label> CallLinkInfo::emitFastPathImpl(CallLinkInfo* callLinkInfo, CCallHelpers& jit, UseDataIC useDataIC, bool isTailCall, ScopedLambda<void()>&& prepareForTailCall)
+void CallLinkInfo::emitFastPathImpl(CallLinkInfo* callLinkInfo, CCallHelpers& jit, UseDataIC useDataIC, bool isTailCall, ScopedLambda<void()>&& prepareForTailCall)
```

Comment:
Since we know whether a tail call is being emitted anytime we call this function, could it be split into two (e.g. `emitTailCallFastPathImpl`, `emitFastPathImpl`)? It would avoid the iffy `nullptr` passing in those cases where `isTailCall = false`.

Reply:
`nullptr` is passed when it is DataOnly CallLinkInfo.